src/main/java/games/strategy/engine/data/Route.java

```diff
@@ -147,21 +147,16 @@ public Territory getStart() {
    */
   public boolean crossesWater() {
     final boolean startLand = !m_start.isWater();
-    boolean overWater = false;
-    final Iterator<Territory> routeIter = m_steps.iterator();
-    Territory terr = null;
-    while (routeIter.hasNext()) {
-      terr = routeIter.next();
-      if (terr.isWater()) {
-        overWater = true;
-      }
-    }
-    if (terr == null) {
+    final boolean overWater = m_steps.stream().anyMatch(Territory::isWater);
+
+    final Territory lastTerritory = m_steps.get(m_steps.size() - 1);
```

Comment:
Won't this throw `IndexOutOfBoundsException` if `m_steps` is empty?

Reply:
Your're correct
Will fix this